### PR TITLE
Update dependencies / build with go 1.19

### PR DESCRIPTION
**Pull request by mrueg**
This change updates all dependencies, switches to yaml.v3 and uses go 1.19 to build.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,11 +3,11 @@ module github.com/xperimental/nextcloud-exporter
 go 1.17
```
Comment (xperimental) [inline go.mod:3]: Can you update this as well with `go mod tidy -go 1.19`?

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
     - name: Setup Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.17
+        go-version: 1.19
```
Comment (xperimental) [inline .github/workflows/package.yml:20]: Seems like I forgot to update this workflow. Would you update it to `setup-go@v3` and usage of the `go.mod` file like the other workflow? I can also do this separately if you like.

**Review by xperimental — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (mrueg) [reply go.mod]: Sure, done!

File: .github/workflows/package.yml (hunk 2 above)
Comment (mrueg) [reply .github/workflows/package.yml]: Good point, changed it. Nice, I wasn't aware of that feature!

File: go.mod (hunk 1 above)
Comment (xperimental) [reply go.mod]: Can you run `go mod tidy`, so that it updates the `go.sum` file as well?

Comment (mrueg) [reply go.mod]: Whoops, missed that. Done!

**Review by xperimental — APPROVED** [review]

**Comment by mrueg** [comment]
@xperimental would you be able to release a new version? There have been a couple of interesting changes since the last release. :)

**Comment by xperimental** [comment]
@mrueg Good idea. I wanted to get #75 into the next release too, but had other things to do the last days. Let's see when there's time.

**Pull request merged**